bokehjs/test/integration/tables.ts

```diff
@@ -20,4 +20,37 @@ describe("DataTable", () => {
     await mouse_click(el)
     await view.ready
   })
+
+  it("should allow sorting with a NanCompare", async () => {
+    const source = new ColumnDataSource({data: {foo: [10, 20, 30, 40], bar: [3.4, NaN, 0, -10]}})
+
+    const foo_col = new TableColumn({field: "foo", title: "Foo", width: 200})
+    const bar_col = new TableColumn({field: "bar", title: "Bar", width: 200, sorter: new NanCompare()})
+    const columns = [foo_col, bar_col]
+
+    const table = new DataTable({source, columns, sortable: true})
+    const {view} = await display(table, [600, 400])
+
+    const el = view.shadow_el.querySelectorAll(".slick-header-column")[2]
+    await mouse_click(el)
+    await view.ready
+  })
+
+  it("should allow sorting with a CustomJSCompare", async () => {
+    const source = new ColumnDataSource({data: {foo: [10, 20, 30, 40], bar: ["a 1", "a 10", "a 100", "a 2"]}})
+
+    const foo_col = new TableColumn({field: "foo", title: "Foo", width: 200})
+
+    const sorter = new CustomJSCompare({code: 'const xn = Number(x.split(" ")[1]); const yn = Number(y.split(" ")[1]); return (xn == yn) ? 0 : (xn < yn) ? -1 : 1'})
```

Comment:
One last change I would suggest is to use multi-line strings (using backticks), which allows to make this snippet more readable:
```suggestion
    const sorter = new CustomJSCompare({
        code: `
            const xn = Number(x.split(" ")[1])
            const yn = Number(y.split(" ")[1])
            return xn == yn ? 0 : (xn < yn ? -1 : 1)
         `,
     })
```
I also changed parentheses in the return statement, because to me the original didn't actually disambiguate anything. In fact they are unnecessary altogether, but I think this is the cleanest version.